src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        checking();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        generateIndexException(index < 0);
```

Comment:
Instead of using a separate `generateIndexException` method for index validation, consider creating a single method that handles all index validations to avoid repeating code.

Reply:
One method has to handle different conditions